x-pack/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -39,6 +48,142 @@ const range1d = { range: { '@timestamp': { gte: 'now-1d' } } };
+      };
+
+      const allResult = await search({
+        index: [indices['apm_oss.transactionIndices']],
+        body: {
+          size: 0,
+          timeout,
+          query: { bool: { filter } },
+          aggs: {
+            current_implementation: compositeAgg('current_implementation'),
+            no_observer_name: compositeAgg('no_observer_name'),
+          },
+        },
+      });
+
+      const noRumResult = await search({
+        index: [indices['apm_oss.transactionIndices']],
+        body: {
+          size: 0,
+          timeout,
+          query: {
+            bool: {
+              filter: [
+                ...filter,
+                { terms: { [AGENT_NAME]: nonRumAgentNames } },
+              ],
+            },
+          },
+          aggs: {
+            no_rum: compositeAgg('no_rum'),
+            no_rum_no_observer_name: compositeAgg('no_rum_no_observer_name'),
+          },
+        },
+      });
+
+      const rumResult = await search({
+        index: [indices['apm_oss.transactionIndices']],
+        body: {
+          size: 0,
+          timeout,
+          query: {
+            bool: {
+              filter: [...filter, { terms: { [AGENT_NAME]: RUM_AGENTS } }],
+            },
+          },
+          aggs: {
+            only_rum: compositeAgg('only_rum'),
+            only_rum_no_observer_name: compositeAgg(
+              'only_rum_no_observer_name'
+            ),
+          },
+        },
+      });
+
+      const allCount = allResult.hits.total.value;
+      const noRumCount = noRumResult.hits.total.value;
+      const rumCount = rumResult.hits.total.value;
+
+      return {
+        aggregated_transactions: {
+          current_implementation: {
+            transaction_count: allCount,
+            expected_metric_document_count:
+              allResult.aggregations?.current_implementation.buckets.length,
+          },
+          no_observer_name: {
+            transaction_count: allCount,
+            expected_metric_document_count:
+              allResult.aggregations?.no_observer_name.buckets.length,
+          },
+          no_rum: {
+            transaction_count: noRumCount,
+            expected_metric_document_count:
+              noRumResult.aggregations?.no_rum.buckets.length,
+          },
+          no_rum_no_observer_name: {
+            transaction_count: noRumCount,
+            expected_metric_document_count:
+              noRumResult.aggregations?.no_rum_no_observer_name.buckets.length,
+          },
+          only_rum: {
+            transaction_count: rumCount,
+            expected_metric_document_count:
+              rumResult.aggregations?.only_rum.buckets.length,
+          },
+          only_rum_no_observer_name: {
+            transaction_count: rumCount,
+            expected_metric_document_count:
+              rumResult.aggregations?.only_rum_no_observer_name.buckets.length,
+          },
+        },
+      };
+    },
+  },
```

Comment:
FWICT the results of the buckets are not being paginated through, so they will always return <= 10k. It might be a (lot) more than that (I've seen 60k+). The script I mentioned also paginates through the buckets (by using `after_key`).